server/middleware/setUpAuthentication.ts

```diff
@@ -50,7 +50,7 @@ export default function setupAuthentication() {
 
   router.get('/sign-in/callback', (req, res, next) =>
     passport.authenticate('oauth2', {
-      successReturnToOrRedirect: req.session.returnTo || '/disclaimer',
+      successReturnToOrRedirect: '/disclaimer',
```

Comment:
all go thru disclaimer which then decides (in one place) whether they can move to "search"